### PR TITLE
Check for llvm-tools before install

### LINKED ISSUE #108948: rust 1.68.0 install fails with `missing llvm-tools` if using external llvm
Close to the end of `dist` target there's:
```
Skipping LlvmTools (aarch64-unknown-linux-gnu): external LLVM                
```
but shortly after `install` fails with:
```
thread 'main' panicked at 'missing llvm-tools', install.rs:215:14                                                                                                                              
```

### PR DESCRIPTION
**Pull request by chaitanyav**
Fixes #108948
@jpalus Please review

### CODE REVIEW

**Comment by albertlarsan68** [comment]
Thanks for the PR!
@bors r+ rollup

**Comment by jpalus** [comment]
Can't give much of a review but I can confirm this PR fixes #108948 and install skips llvm tools installation now:
```
Skipping LlvmTools (aarch64-unknown-linux-gnu): external LLVM                                  
skipping llvm-tools stage2 (aarch64-unknown-linux-gnu): external LLVM 
```

File: src/bootstrap/install.rs [hunk 1]
```diff
@@ -210,10 +210,13 @@ install!((self, builder, _config),
         }
     };
     LlvmTools, alias = "llvm-tools", Self::should_build(_config), only_hosts: true, {
-        let tarball = builder
-            .ensure(dist::LlvmTools { target: self.target })
-            .expect("missing llvm-tools");
-        install_sh(builder, "llvm-tools", self.compiler.stage, Some(self.target), &tarball);
+        if let Some(tarball) = builder.ensure(dist::LlvmTools { target: self.target }) {
```
Comment (ozkanonur) [inline src/bootstrap/install.rs:213]: I don't think this approach was right.

See https://github.com/rust-lang/rust/pull/109196#discussion_r1138336815

Comment (chaitanyav) [reply src/bootstrap/install.rs:213]: @ozkanonur Thank you for fixing.

**Pull request merged**